### PR TITLE
Add check for extensions min Java

### PR DESCRIPTION
**Pull request by ia3andy**
```
~/D/quarkus-update> qss create app -e --java=11 -x=resteasy-reactive -x=PrimeFaces -S=3.0 hello-4
[ERROR] ❗  Some extensions are not compatible with the selected Java version (11):
 io.quarkiverse.primefaces:quarkus-primefaces (min: 17)
 ```

### CODE REVIEW

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java [hunk 1]
```diff
@@ -74,15 +65,15 @@ public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws
 
         List<Extension> extensionsToAdd = computeRequiredExtensions(invocation.getExtensionsCatalog(), extensionsQuery,
                 invocation.log());
-
-        ExtensionCatalog mainCatalog = invocation.getExtensionsCatalog(); // legacy platform initialization
         final List<ExtensionCatalog> extensionOrigins;
+        ExtensionCatalog mainCatalog = invocation.getExtensionsCatalog(); // legacy platform initialization
+        final Integer javaVersion = Integer.parseInt(invocation.getStringValue(JAVA_VERSION));
```
Comment (aloubyansky) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java:70]: Perhaps check for null?

Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java]: Fixed

**Comment by ia3andy** [comment]
@aloubyansky could you have a look? I think the CI is failing on something else..

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java [hunk 2]
```diff
@@ -244,4 +234,25 @@ private static void addOrigins(final List<ExtensionOrigins> extOrigins, Extensio
...
     }
+
+    private void checkMinimumJavaVersion(String javaVersionString, List<Extension> extensions) throws QuarkusCommandException {
+        final List<Extension> incompatibleExtensions = new ArrayList<>();
+        final int javaVersion = javaVersionString == null ? CreateProjectHelper.DEFAULT_JAVA_VERSION
+                : Integer.parseInt(javaVersionString);
+        for (Extension extension : extensions) {
+            Integer extMinJavaVersion = getMinimumJavaVersion(extension);
+            if (extMinJavaVersion != null
+                    && CreateProjectHelper.JAVA_VERSIONS_LTS.contains(extMinJavaVersion)
```
Comment (aloubyansky) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java:245]: What is the reason for checking whether it's an LTS?

Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java:245]: This was a way to make sure the value is valid and not a mistake, do you see something else than a LTS in there? Also maybe we should enfore extension to be built on LTS no?

Comment (aloubyansky) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java:245]: I don't think we should be enforcing it.

**Review by aloubyansky — APPROVED** [review]
Thanks @ia3andy

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java (hunk 2 above)
Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java]: Fixed

**Pull request merged**